### ISSUE TITLE
Implementing Android Wear 2.0 Ambient Mode Support.

### ISSUE THREAD

**Issue by ThomasJFR** [NativeScript/NativeScript#5963]
I am trying to implement android wear ambient mode support on a {N} app, but can't find any way to override ambient mode methods (e.g onEnterAmbient(), onExitAmbient()...). I have tried creating a custom class that extends FragmentActivity and have overridden the methods there but I am getting an error: 

`System.err: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.nativescript.workertest/vigil.pin.Activity}: com.tns.NativeScriptException:
System.err: Calling js method onCreate failed
System.err:
System.err: Error: JNI Exception occurred (SIGABRT).
` 

I am wondering if there is a way to get access to these methods and override them.

Version numbers:
- CLI: 4.1.0
- Runtime(s): Android 4.2.0

FragmentActivity extended class: 

```

import {setActivityCallbacks, AndroidActivityCallbacks} from "ui/frame";

declare var android: any;

@JavaProxy("my.new.Activity")
class Activity extends android.v4.support.appy.FragmentActivity {
    private _callbacks: AndroidActivityCallbacks;

    protected onCreate(savedInstanceState/*: android.os.Bundle*/): void {
        if (!this._callbacks) {
            setActivityCallbacks(this);
        }

        this._callbacks.onCreate(this, savedInstanceState, super.onCreate);
    }

    protected onStart(): void {
        this._callbacks.onStart(this, super.onStart);
    }

    protected onStop(): void {
        this._callbacks.onStop(this, super.onStop);
    }

    
    protected onDestroy(): void {
        this._callbacks.onDestroy(this, super.onDestroy);
    }

    public onKeyDown( keyCode, event ): boolean {
        console.log('Key down: ' + keyCode);
        return true;
    }


    
    public getAmbientCallback() {
        return new Activity.MyAmbientCallback();
    }   


    private static MyAmbientCallback = class extends android.support.wear.ambient.AmbientModeSupport.AmbientCallback {
        public onEnterAmbient(ambientDetails) {
            console.log("Entering ambient mode");
        }
    
        public onExitAmbient() {
            console.log("Exiting ambient mode");

            }
    
        public onUpdateAmbient() {
            // Update content
        }
    }
```
Note: I have added the JavaProxy name to the Android Manifest activity declaration. I have included wearable dependencies in the app.gradle file:

```
  compile 'com.android.support:wear:27.0.0'
  compile 'com.google.android.gms:play-services-wearable:11.2.+'
  compile 'com.google.android.support:wearable:2.1.0+'
  provided 'com.google.android.wearable:wearable:2.1.0+'
```

**Comment by NickIliev** [comment]
@ThomasJFR try the following data conversions (marshalling) syntax to overwrite the methods in your callback.

```JS
let MyAmbientCallback = android.support.wear.ambient.AmbientModeSupport.AmbientCallback.extend({
    onEnterAmbient(ambientDetails) {
        console.log("Entering ambient mode");
    },

    onExitAmbient() {
        console.log("Exiting ambient mode");
    },

    onUpdateAmbient() {
        // Update content
    }
});
```

**Comment by ThomasJFR** [comment]
Thanks! That's cleared up the issue I was having previously. However, I'm now getting the issue

`TypeError: Cannot read property 'AmbientCallback' of undefined`

Seems that `android.support.wear.ambient` isn't defined (I know for sure that `android.support.wear` is, because I was using other wear methods earlier on). I assume this is because of dependencies, but I can't seem to figure out what's causing the issue. 

I've added my dependencies above - if you're able to see anything out of the ordinary, I'd really appreciate you letting me know. Thanks in advance!

**Comment by NickIliev** [comment]
@ThomasJFR based [on the Android docs for `AmbienModeSupport` ](https://developer.android.com/reference/android/support/wear/ambient/AmbientModeSupport)the class is available for API 27.0.1 and above.

```
AmbientModeSupport
added in version 27.1.0 
belongs to Maven artifact 
com.android.support:wear:28.0.0-alpha1
```

Meaning you will ahve to add the native dependency to `com.android.support:wear:28.0.0-alpha1` or later version in your `app.gradle`. Example for adding native depedency in app.gradle [here](https://github.com/NickIliev/nativescript-ng-cosmos/blob/master/app/App_Resources/Android/app.gradle#L1-L3).
